bundles/org.openhab.binding.kvv/src/main/java/org/openhab/binding/kvv/internal/KVVStopHandler.java

```diff
@@ -0,0 +1,177 @@
+            return;
+        }
+
+        // creating channels
+        if (createChannels) {
+            final List<Channel> channels = new ArrayList<Channel>();
+            for (int i = 0; i < bridgeHandler.getBridgeConfig().maxTrains; i++) {
+                channels.add(ChannelBuilder.create(new ChannelUID(this.thing.getUID(), "train" + i + "-name"), "String")
+                        .build());
+                channels.add(ChannelBuilder
+                        .create(new ChannelUID(this.thing.getUID(), "train" + i + "-destination"), "String").build());
+                channels.add(ChannelBuilder.create(new ChannelUID(this.thing.getUID(), "train" + i + "-eta"), "String")
+                        .build());
+            }
+            this.updateThing(this.editThing().withChannels(channels).build());
+        }
+        this.pollingJob = this.scheduler.scheduleWithFixedDelay(new UpdateTask(bridgeHandler, this.config), 0,
+                bridgeHandler.getBridgeConfig().updateInterval, TimeUnit.SECONDS);
+
+        this.bridgeHandler = bridgeHandler;
+    }
+
+    @Override
+    public void dispose() {
+        if (this.pollingJob != null) {
+            this.pollingJob.cancel(false);
+        }
+    }
+
+    /**
+     * Updates the local list of departures.
+     *
+     * @param departures the new list of departures
+     */
+    private synchronized void setDepartures(final DepartureResult departures, final int maxTrains) {
+        int i = 0;
+        for (; i < departures.departures.size(); i++) {
+            this.updateState(new ChannelUID(this.thing.getUID(), "train" + i + "-name"),
+                    new StringType(departures.departures.get(i).route));
+            this.updateState(new ChannelUID(this.thing.getUID(), "train" + i + "-destination"),
+                    new StringType(departures.departures.get(i).destination));
+            String eta = departures.departures.get(i).time;
+            if (eta.equals("0")) {
+                eta += " min";
+            }
+            this.updateState(new ChannelUID(this.thing.getUID(), "train" + i + "-eta"), new StringType(eta));
+        }
+        for (; i < maxTrains; i++) {
+            this.updateState(new ChannelUID(this.thing.getUID(), "train" + i + "-name"), new StringType(""));
+            this.updateState(new ChannelUID(this.thing.getUID(), "train" + i + "-destination"), new StringType(""));
+            this.updateState(new ChannelUID(this.thing.getUID(), "train" + i + "-eta"), new StringType(""));
+        }
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        if (command == RefreshType.REFRESH) {
+            init(false);
+        }
+    }
+
+    /**
+     * Holds a single {@link TimerTask} to fetch the latest departure data.
+     *
+     * @author Maximilian Hess - Initial contribution
+     *
+     */
+    @NonNullByDefault
+    public class UpdateTask extends TimerTask {
```

Comment:
This class is not needed. You can just create a method `updateTask` without parameters and pass it to the scheduler as `this::updateTask`